icicle-source/src/Icicle/Dictionary/Data.hs

```diff
@@ -54,121 +50,66 @@ import           P
 
 
 data Dictionary =
-  Dictionary
-  { dictionaryEntries   :: [DictionaryEntry]
-  , dictionaryFunctions :: [ (Name Variable
-                           , ( ST.FunctionType Variable
-                             , Function (ST.Annot SourcePos Variable) Variable)) ] }
-  deriving (Eq, Show)
-
-data DictionaryEntry =
-  DictionaryEntry
-  { dictionaryEntryAttribute  :: Attribute
-  , dictionaryEntryDefinition :: Definition
-  , dictionaryEntryNamespace  :: Namespace }
-  deriving (Eq, Show)
-
-data Definition =
-    ConcreteDefinition Encoding Tombstones (ConcreteKey AnnotSource Variable)
-  | VirtualDefinition  Virtual
-  deriving (Eq, Show)
-
-type Tombstones = Set Text
-
--- | A parsed and typechecked source program.
-newtype Virtual = Virtual {
-    unVirtual :: QueryTop (ST.Annot SourcePos Variable) Variable
-  } deriving (Eq, Show)
+  Dictionary {
+      dictionaryInputs :: Map InputId DictionaryInput
+    , dictionaryOutputs :: Map OutputId DictionaryOutput
+    , dictionaryFunctions :: [DictionaryFunction]
+    } deriving (Eq, Show)
```

Comment:
🗒 dictionary has been rejigged to separate inputs from outputs and put them in maps to ensure ordering